lib/session.js

```diff
@@ -26,18 +27,85 @@ function Session(host, auth, options) {
   this._callbacks = [];
 
   this._debug = debug('nodemiral:sess:' + host);
+
+  // null means it's not set
+  self._ssh_config = null;
+
+  this._getSshConfigInfo(this._options.config, function (err, config) {
+    if(!err) {
+      self._ssh_config = config;
+      return;
+    } else if(self._options.config) {
+      throw err;
+    } else {
+      self._ssh_config = {};
+    }
+  });
 }
 
+Session.prototype._getSshConfigInfo = function(file, callback) {
+  var file = file || '~/.ssh/config';
+  var host = this._host;
+
+  fs.readFile(file, {encoding: 'utf8'}, function(err, content) {
+    if(err) {
+      return callback(err);
+    }
+
+    var connInfo = {};
+    var config = sshConfig.parse(content);
+
+    for(var i=0; i<config.length; ++i) {
+      var conf = config[i];
+
+      if(conf.Host === '*') {
+        if(conf.HostName) {
+          connInfo.host = connInfo.host || conf.HostName;
+        }
+
+        if(conf.User) {
+          connInfo.username = connInfo.username || conf.User;
+        }
+
+        if(conf.IdentityFile) {
+          connInfo.privateKey = connInfo.privateKey || conf.IdentityFile;
```

Comment:
We may need to a `fs.ReadFile` for the IdentifyFile. `privateKey` is the key it self, not the path. 
So, you may need to change the code a bit or use generators :)